test/common_test/joint_features.cc

```diff
@@ -1296,9 +1296,9 @@ TYPED_TEST(JointFeaturesAttachDetachTest, JointAttachMultiple)
       gz::math::Vector3d body3LinearVelocity =
           gz::math::eigen3::convert(frameDataModel3Body.linearVelocity);
 
-      EXPECT_NEAR(dt * (numSteps) * -10, body1LinearVelocity.Z(), 1e-3);
+      EXPECT_NEAR(dt * (numSteps) * -10, body1LinearVelocity.Z(), 3e-3);
```

Comment:
nit: you can make these checks relative to the velocity before stepping to make it more robust:
```
    frameDataModel1Body = model1Body->FrameDataRelativeToWorld();
    const double preStepBody1LinearVelocityZ = frameDataModel1Body.linearVelocity.z();
    ...
    for (std::size_t i = 0; i < numSteps; ++i)
    {
      world->Step(output, state, input);
    }
    {
      frameDataModel1Body = model1Body->FrameDataRelativeToWorld();
      ...
      // Expect the middle box to be still as it is already at rest.
      // Expect the two side boxes to fall away.

      EXPECT_NEAR(preStepBody1LinearVelocityZ + dt * (numSteps) * -10, frameDataModel1Body.linearVelocity.z(), 1e-3);
```
That way, you should be able to keep using the original conservative threshold of 1e-3 for this check.

Reply:
updated checks to take into account the pre-step body velocity. [8ea3e67](https://github.com/gazebosim/gz-physics/pull/600/commits/8ea3e67e90177a28a5e6ed31da02dbec7bb93415)